Amplify/Categories/Auth/Request/AuthConfirmSignUpRequest.swift

```diff
@@ -0,0 +1,41 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+public struct AuthConfirmSignUpRequest: AmplifyOperationRequest {
+
+    public let username: String
+
+    public let code: String
+
+    public var options: Options
+
+    public init(username: String, code: String, options: Options) {
+        self.username = username
+        self.code = code
+        self.options = options
+    }
+}
+
+public extension AuthConfirmSignUpRequest {
+
+    struct Options {
+
+        public let forceAliasCreation: Bool
+        public let validationData: [String: String]?
+        public let metaData: [String: String]?
+
+        public init(forceAliasCreation: Bool = false,
+                    validationData: [String: String]? = nil,
+                    metaData: [String: String]? = nil) {
```

Comment:
For some of these request objects, the params are somewhat self documenting. From having used sign in systems, I know what and current password, new password, and confirmation code are.

These are different. We should get some Apple doc to tell us the intention of `forceAliasCreation`, `validationData`, and `metaData`.

Reply:
Will be handled in a separate PR for comments. Added a TODO.